### PR TITLE
Add corpus metadata sidecars, benchmark pipeline, and tests

### PR DESCRIPTION
**Pull request by cnkang**
## Summary by Sourcery

Introduce a benchmarked HTML corpus with metadata sidecars and a unified reporting pipeline for measuring converter quality and regression, and wire it into local tooling and CI gates.

New Features:
- Add a corpus benchmark runner that executes the converter over the HTML corpus, computes quality and latency metrics, and emits a unified JSON report with per-fixture results.
- Introduce report comparison tooling that evaluates baseline versus current corpus metrics against configurable thresholds and produces a machine-readable verdict.
- Add a formatter that generates markdown benchmark summaries for inclusion in PRs from the unified report.
- Define a structured corpus with versioning, page-type classification, and metadata sidecars for each HTML fixture, including a designated failure corpus and new boilerplate-heavy examples.

Enhancements:
- Extend the corpus validation script to verify corpus-version metadata, enforce presence and schema of .meta.json sidecars, and check coverage for page types and failure fixtures.
- Add Makefile targets to run the corpus benchmark, compare reports, and generate PR summaries, integrating them into the existing tooling workflow.

CI:
- Add a Corpus Benchmark Gate job in CI and nightly perf workflows that runs the corpus benchmark, compares against the baseline, uploads reports and examples, and ensure it triggers on relevant path changes including corpus tooling.
- Add regression tests to guard CI workflow configuration so corpus tooling changes continue to trigger the benchmark gate.

Documentation:
- Document the benchmark corpus design, schema, thresholds, CI integration, and how to run benchmarks locally in a new testing guide.

Tests:
- Add extensive property-based and unit tests for the corpus benchmark, report schema, comparison logic, PR summary formatting, and error-handling paths.

## Release Notes

* **New Features**
  * Corpus benchmarking infrastructure for performance tracking with automated baseline comparison and pass/warn/fail verdicts
  * Performance metrics including token reduction, latency percentiles, and fallback rate

* **Tests**
  * Added comprehensive test corpus with 33+ fixtures covering edge cases, encoding, and malformed HTML scenarios
  * Added performance benchmark validation tests

* **Documentation**
  * Added benchmark corpus documentation with schema specifications and metrics interpretation

* **Chores**
  * Integrated corpus benchmarking into CI pipeline with artifact reporting
  * Added performance tooling for measurement and comparison

### CODE REVIEW

**Comment by cnkang** [comment]
@codex review

**Comment by cnkang** [comment]
@sourcery-ai review

**Comment by cnkang** [comment]
@sourcery-ai review

File: tools/perf/run_corpus_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+        # execvp — no shell expansion, no injection vector.  Both paths are
+        # validated with os.path.isfile() above.  shlex.escape() is NOT
+        # appropriate here; it is designed for shell=True string concatenation
+        # and would corrupt paths containing spaces or special characters.
+        result = subprocess.run(  # sourcery skip: avoid-subprocess-run
+            [converter_bin, html_path],  # noqa: S603 — args are validated paths
+            capture_output=True,
+            text=True,
+            timeout=30,
+        )
```
[the remark this answers is not in the document]
Comment (cnkang) [reply tools/perf/run_corpus_benchmark.py]: Security: list-form invocation (shell=False) passes args directly to execvp — no shell expansion, no injection vector.  Both paths are validated with os.path.isfile() above.  shlex.escape() is NOT appropriate here; it is designed for shell=True string concatenation and would corrupt paths containing spaces or special characters.

**Comment by cnkang** [comment]
@sourcery-ai review
